python/arcticdb/version_store/library.py

```diff
@@ -1288,6 +1288,73 @@ def get_description(self, symbol: str, as_of: Optional[AsOf] = None) -> SymbolDe
             date_range=info["date_range"],
         )
 
+    def get_description_batch(
+        self, symbols: List[Union[str, ReadRequest]]
+    ) -> List[SymbolDescription]:
+        """
+        Returns descriptive data for a list of ``symbols``.
+
+        Parameters
+        ----------
+        symbols : List[Union[str, ReadRequest]
+            List of symbols to read.
+            Params columns, date_range and query_builder from ReadRequest are not used
+
+        Returns
+        -------
+        List[SymbolDescription]
+            A list of the descriptive data, whose i-th element corresponds to the i-th element of the ``symbols`` parameter.
+
+        See Also
+        --------
+        SymbolDescription
+            For documentation on each field.
+        """
+        symbol_strings = []
+        as_ofs = []
+        def handle_read_request(s_):
+            symbol_strings.append(s_.symbol)
+            if s_.date_range is not None:
+                warnings.warn("Read batch metadata does not make use of of the data_range field", SyntaxWarning, stacklevel=2)
+            if s_.columns is not None:
+                warnings.warn("Read batch metadata does not make use of the columns field", SyntaxWarning, stacklevel=2)
+            if s_.query_builder is not None:
+                warnings.warn("Read batch metadata does not make use of the query field", SyntaxWarning, stacklevel=2)
+            as_ofs.append(s_.as_of)
+
+        def handle_symbol(s_):
+            symbol_strings.append(s_)
+            as_ofs.append(None)
+
+        for s in symbols:
+            if isinstance(s, str):
+                handle_symbol(s)
+            elif isinstance(s, ReadRequest):
+                handle_read_request(s)
+            else:
+                raise ArcticInvalidApiUsageException(
+                    f"Invalid symbol type s=[{s}] type(s)=[{type(s)}]. Only [str] and [ReadRequest] are supported."
```

Comment:
Minor: I would say "Unsupported item in the symbols argument..."

`symbol type` may have other connotations.